Cocktails9/Cocktails9/Model/CocktailDetailViewModel.swift

```diff
@@ -0,0 +1,101 @@
+import SwiftUI
+
+class CocktailDetailViewModel: ObservableObject {
+    @Published var cocktail: Cocktail? = nil
+    @Published var isLoading = false
+    @Published var errorMessage: String? = nil
+    
+    func fetchCocktailDetails(id: String) {
+        isLoading = true
+        errorMessage = nil
+        
+        Task {
+            do {
+                let url = URL(string: "https://www.thecocktaildb.com/api/json/v1/1/lookup.php?i=\(id)")!
+                let (data, _) = try await URLSession.shared.data(from: url)
+                let decodedResponse = try JSONDecoder().decode(CocktailResponse.self, from: data)
+                
+                if let cocktail = decodedResponse.drinks?.first {
+                    self.cocktail = cocktail
+                } else {
+                    self.errorMessage = "Cocktail details not found."
+                }
+            } catch {
+                self.errorMessage = "Failed to load cocktail details."
+            }
+            isLoading = false
+        }
+    }
+}
+
+// Data models to decode the response
+struct CocktailResponse: Codable {
+    let drinks: [Cocktail]?
+}
+
+struct Cocktail: Codable, Identifiable {
+    var id: String { strDrink }
+    
+    let strDrink: String
```

Comment:
You can use coding keys to keep variables more readable, `drink` instead of `strDrink`